google/cloud/storage/examples/storage_grpc_samples.cc

```diff
@@ -74,11 +74,59 @@ void GrpcClientWithProject(std::string project_id) {
 }
 //! [grpc-client-with-project]
 
+//! [report-transport]
+void GrpcReportTransport(std::string const& config,
+                         std::string const& bucket_name) {
```

Comment:
Should this sample take a `gcs::Client& client` instead of a `std::string const& config`?

`GrpcClientWithDP()` already says how to make a DP client.

Reply:
Also, should there be a comment that says why we cannot just: `std::cout << config << "\n";`

Is either of the following true?
- even if you ask for DP, it might fall back to non-DP
- sometimes when you use the regular gRPC endpoint it uses DP anyway

Explaining the motivation later in whatever documentation will reference this sample is fine.

Reply:
> even if you ask for DP, it might fall back to non-DP

yes.

I included some comments motivating the example.